plugins/woocommerce-admin/client/products/product-validation.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { ProductStatus, ProductType, Product } from '@woocommerce/data';
+
+export const validate = (
+	values: Partial< Product< ProductStatus, ProductType > >
+) => {
+	const errors: {
+		[ key: string ]: string;
+	} = {};
+	if ( ! values.name?.length ) {
+		errors.name = __( 'This field is required.', 'woocommerce' );
+	}
+
+	if ( values.name && values.name.length > 120 ) {
+		errors.name = __(
+			'Please enter a product name shorter than 120 characters.',
+			'woocommerce'
+		);
+	}
+
+	if ( values.regular_price && ! /^[0-9.,]+$/.test( values.regular_price ) ) {
+		errors.regular_price = __(
+			'Please enter with one monetary decimal point without thousand separators and currency symbols.',
```

Comment:
I guess we're trying to keep this consistent for easier translation, right?  I think adding `a price` may make this slightly more clear.

```suggestion
			'Please enter a price with one monetary decimal point without thousand separators and currency symbols.',
```

Although, I would expect a price with thousands separator or currency to work and be sanitized.

Reply:
Nice catch! Yes, the price is currently sanitized. I added the validation just to be more defensive. It wouldn't hit the validation because I'm sanitizing it in the onChange method. I fixed the text in the commit dd6622e